operator/operator.go

```diff
@@ -160,6 +161,75 @@ func (operator *Operator) handleBlocks() error {
 	return nil
 }
 
+func (operator *Operator) handleChallenges() error {
+	challenges := make(chan *bindings.ErdstallChallenged)
+	sub, err := operator.contract.WatchChallenged(nil, challenges, nil, nil)
+	if err != nil {
+		return fmt.Errorf("creating challenge subcription: %w", err)
+	}
+	defer sub.Unsubscribe()
+
+	for {
+		select {
+		case err := <-sub.Err():
+			return fmt.Errorf("subscription error: %w", err)
+
+		case c := <-challenges:
```

Comment:
You could move the whole body of this `case` into its own method `handleChallenge(c *bindings.ErdstallChallenged) error` on the `*Operator` which would help readability because you save 2 levels of indentation. Then you can also replace all `log.Errorf(...); continue` statements with just `return fmt.Errorf(...)` and do the `log.Errorf(...); continue` here where you call the single challenge handler.

Reply:
done